elyra/pipeline/component.py

```diff
@@ -275,73 +236,158 @@ class ComponentReader(LoggingConfigurable):
     def type(self) -> str:
         return self.type
 
+    @property
+    def base_type(self):
+        return self.type
```

Comment:
what's the difference between `type` and `base_type`?

Reply:
Good question. `type` corresponds to either `url`, `directory`, or `filename` (and whichever other reader classes we introduce eventually). It is used to assign the appropriate reader for that class since they each have their own considerations when accessing locations. 

`base_type` is one of either `filename` or `url` (where directory's `base_type` is `filename` and an eventual github registry's `base_type`, e.g., would be `url`). This is what is assigned to the component as it's `source_type`. The reason for this is that the `source_type` field is used during processing of the KFP components. We look up the component and it's `source_type` in the registry cache, then the component is loaded via `load_component(url='someurl')` or `load_component(filename='somefile')`

Reply:
I see. So perhaps we could change `type` to `location_type` and `base_type` to `resource_type` with the explanation relative to the new names.  `location_type` indicates how to locate resources while `resource_type` indicates how to read the resource.